test/jdk/tools/jpackage/windows/JpackageTest.java

```diff
@@ -0,0 +1,172 @@
+    }
+
+    /**
+     * Sets up the prerequisites for executing the test - Compile the third
+     * party application launcher source - Create a jar of third party
+     * application launcher - Create a windows installer package of third party
+     * application launcher
+     */
+    private void setUp() throws Throwable {
+
+        CompilerUtils.compile(srcDir.resolve(SRC_FILE), classesDir);
+        logger.info("Compiled Successfully");
+
+        Manifest manifest = new Manifest();
+        manifest.getMainAttributes().put(Attributes.Name.MANIFEST_VERSION,
+                "1.0");
+        manifest.getMainAttributes().put(Attributes.Name.MAIN_CLASS,
+                APPLICATION_NAME);
+
+        JarUtils.createJarFile(Paths.get(JAR_FILE), manifest, classesDir,
+                classesDir.resolve(CLASS_FILE));
+        logger.info("Jar Created Successfully");
+
+        ProcessTools
+                .executeCommand(JPACKAGE, "--input", CWD.toString(), "--name",
+                        APPLICATION_NAME, "--main-jar", JAR_FILE,
+                        "--main-class", APPLICATION_NAME, "--type", "exe")
+                .shouldHaveExitValue(0);
+        logger.info("Package Successfully");
+
+        ProcessTools.executeCommand(INSTALLER).shouldHaveExitValue(0);
+        logger.info("Installed Successfully");
+    }
+
+    /**
+     * Execute the test - Start the third party application launcher - Check
+     * whether the termination of third party application launcher also
+     * terminating the launched third party application - If third party
+     * application is not terminated the test is successful else failure
+     */
+    private void run() throws Throwable {
+        ProcessTools.executeCommand(EXECUTABLE_APPLICATION)
+                .shouldHaveExitValue(0);
+        logger.info("Launched Successfully");
+
+        long processId = readProcessId();
+        logger.info(
+                "Successfully read processid from file " + processIdFilePath);
+        logger.info("Process Id is " + processId);
+
+        Optional<ProcessHandle> processHandle = ProcessHandle.of(processId);
+        boolean isAlive = processHandle.isPresent()
+                && processHandle.get().isAlive();
+        if (isAlive) {
+            logger.info("Test Successful");
+        } else {
+            logger.info("Test failed");
+            throw new RuntimeException(
+                    "Test failed: Third party software is terminated");
+        }
+    }
+
+    /**
+     * Helper function to read processId from file
+     */
+    private long readProcessId() throws Throwable {
+
+        Path path = Paths.get(processIdFilePath);
+        long processId = Long.parseLong(Files.readAllLines(path).get(0));
+        return processId;
+    }
+
+    /**
+     * Kill if any regedit.exe is running before launching
+     */
+    private void closeThirdPartyApplication() throws Throwable {
+        Runtime.getRuntime().exec("taskkill /F /IM regedit.exe");
```

Comment:
I guess this will kill all regedit processes including those not started by the test. This doesn't seem right.